### PR TITLE
Create Custom Filter for the AsyncAppender

### PR DESCRIPTION
**Pull request by krankkkk**
Implements One part of the changes for #287
The AsyncAppender is wrapper in an conditional Filter, that only passes the logevents on if an Environment property "TJ_APPENDER" is set.

Note: this means we now also need an compile-time dependency on the log4j2-impl

### CODE REVIEW

**Comment by Zabuzard** [comment]
Is this a draft or is it ready already? cause its failing checks.

File: application/src/main/java/org/togetherjava/tjbot/logging/ConditionalAppender.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import org.apache.logging.log4j.core.config.Property;
+import org.apache.logging.log4j.core.config.plugins.*;
+import org.apache.logging.log4j.core.config.plugins.validation.constraints.Required;
+import org.apache.logging.log4j.core.filter.AbstractFilterable;
+
+import java.io.Serializable;
+
+@Plugin(name = "ConditionalAppender", category = Core.CATEGORY_NAME,
+        elementType = Appender.ELEMENT_TYPE)
+public final class ConditionalAppender extends AbstractAppender {
```
Comment (Zabuzard) [inline application/src/main/java/org/togetherjava/tjbot/logging/ConditionalAppender.java:16]: Javadoc please

File: application/src/main/java/org/togetherjava/tjbot/logging/ConditionalAppender.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+    public static final String LOGGING_FLAG = "TJ_APPENDER";
```
Comment (Zabuzard) [inline application/src/main/java/org/togetherjava/tjbot/logging/ConditionalAppender.java:18]: Javadoc.

File: application/src/main/java/org/togetherjava/tjbot/logging/ConditionalAppender.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+    private final AppenderRef ref;
+    private final Configuration map;
+
+    private ConditionalAppender(final String name, final Filter filter,
+            final Layout<? extends Serializable> layout, final boolean ignoreExceptions,
+            final Property[] properties, AppenderRef ref, Configuration map) {
```
Comment (Zabuzard) [inline application/src/main/java/org/togetherjava/tjbot/logging/ConditionalAppender.java:25]: NotNull on params

File: application/src/main/java/org/togetherjava/tjbot/logging/ConditionalAppender.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+            this.map.getAppender(this.ref.getRef()).append(event);
+        }
+    }
+
+    @PluginBuilderFactory
+    public static Builder newBuilder() {
+        return new Builder();
+    }
+
+    public static class Builder<B extends Builder<B>> extends AbstractFilterable.Builder<B>
```
Comment (Zabuzard) [inline application/src/main/java/org/togetherjava/tjbot/logging/ConditionalAppender.java:44]: Javadoc, final

**Review by Zabuzard — CHANGES_REQUESTED** [review]

**Comment by krankkkk** [comment]
Apologies was intended as a draft, can now be reviewed,
Checks seem to be failing because gradle on Github-Workers cannot find log4j2, but I cannot replicate the issue locally

File: application/src/main/java/org/togetherjava/tjbot/logging/FlaggedFilter.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+
+/**
+ * A custom Filter for Log4j2, which only lets an event pass through if a Logging Flag is set in the
+ * environment. Intended to be used for local development for devs do not want to also run the
+ * logviewer project. No errors in console or Log should appear, if the Flag is not set and the
+ * logviewer is not running.
+ */
+@Plugin(name = "FlaggedFilter", category = Core.CATEGORY_NAME, elementType = Filter.ELEMENT_TYPE)
+public class FlaggedFilter extends AbstractFilter {
```
Comment (Zabuzard) [inline application/src/main/java/org/togetherjava/tjbot/logging/FlaggedFilter.java:20]: final

File: application/src/test/java/org/togetherjava/tjbot/logging/FilterTest.java [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+import org.apache.logging.log4j.core.Filter;
+import org.apache.logging.log4j.core.LogEvent;
+import org.apache.logging.log4j.core.impl.Log4jLogEvent;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+class FilterTest {
```
Comment (Zabuzard) [inline application/src/test/java/org/togetherjava/tjbot/logging/FilterTest.java:11]: final

File: application/src/test/java/org/togetherjava/tjbot/logging/FilterTest.java [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+    FlaggedFilter filter;
+    LogEvent event;
```
Comment (Zabuzard) [inline application/src/test/java/org/togetherjava/tjbot/logging/FilterTest.java:14]: private

**Review by Zabuzard — CHANGES_REQUESTED** [review]

File: application/src/main/java/org/togetherjava/tjbot/logging/FlaggedFilter.java (hunk 5 above)
Comment (krankkkk) [reply application/src/main/java/org/togetherjava/tjbot/logging/FlaggedFilter.java:20]: Cannot be final, else Mockito can't subclass it

**Review by Zabuzard — APPROVED** [review]

**Review by Tais993 — APPROVED** [review]

**Pull request merged**